### PR TITLE
Prioritize disk.DiskSizeGb over disk.InitializeParams.DiskSizeGb

### PR DESCRIPTION
**Pull request by seanrmilligan**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
```

### CODE REVIEW

**Comment by seanrmilligan** [comment]
Note to self to fix the whitespace alignment of added code.

**Comment by seanrmilligan** [comment]
"disk_size_gb2" needs a better name, but I left it as a placeholder so we could have that conversation in the review.

**Review by melinath — COMMENTED** [review]
FYI tests are crashing due to `google_compute_region_instance_template` reusing the flatteners and expanders from `google_compute_instance_template`. Any additional fields would also need to be added there.

File: mmv1/third_party/terraform/services/compute/resource_compute_instance_template.go.erb [hunk 1]
```diff
@@ -1341,10 +1345,13 @@ func flattenDisk(disk *compute.AttachedDisk, configDisk map[string]any, defaultP
 		diskMap["disk_type"] = disk.InitializeParams.DiskType
 		diskMap["disk_name"] = disk.InitializeParams.DiskName
 		diskMap["labels"] = disk.InitializeParams.Labels
-		// The API does not return a disk size value for scratch disks. They can only be one size,
-		// so we can assume that size here.
-		if disk.InitializeParams.DiskSizeGb == 0 && disk.Type == "SCRATCH" {
+		// The API does not return a disk size value for scratch disks. They are largely only one size,
+		// so we can assume that size here. Prefer disk.DiskSizeGb over the deprecated
+		// disk.InitializeParams.DiskSizeGb.
+		if disk.DiskSizeGb == 0 && disk.InitializeParams.DiskSizeGb == 0 && disk.Type == "SCRATCH" {
```
Comment (seanrmilligan) [inline mmv1/third_party/terraform/services/compute/resource_compute_instance_template.go.erb:1351]: Possible logic bug, past and present: If the disk is not of type SCRATCH, we move past the if into the else-if and else blocks. I can put a guard here wrapping this entire section checking for SCRATCH, but it more closely emulates the existing code to not do so.

Comment (trodge) [reply mmv1/third_party/terraform/services/compute/resource_compute_instance_template.go.erb:1351]: This looks correct as-is to me. If the disk type is not `SCRATCH` and both `DiskSizeGb` fields are 0 we can set `disk_size_gb` to 0.

**Review by trodge — APPROVED** [review]

**Pull request merged**